event_model/__init__.py

```diff
@@ -1045,6 +1045,73 @@ def unpack_datum_page(datum_page):
         yield datum
 
 
+def rechunk_event_pages(event_pages, chunk_size):
+    remainder = chunk_size
+    chunk_list = []
+
+    for page in event_pages:
+        new_chunks = page_chunks(page, chunk_size, remainder)
+        for chunk in new_chunks:
+            remainder -= len(chunk['uid'])
+            chunk_list.append(chunk)
+            if remainder == 0:
+                yield merge_event_pages(chunk_list)
+                remainder = chunk_size
+    if chunk_list:
+        yield merge_event_pages(chunk_list)
+
+    def page_chunks(page, chunk_size, remainder):
```

Comment:
Any reason not to make this a top-level private function `_page_chunks`? That would enable us to test it separately. In-lining in a good way to do closures, which I don't think we need here, or to ensure that something definitely isn't exposed to users, which doesn't seem like an issue here either.

Reply:
The reason I planned to nest it because I thought, people shouldn't be using it directly. `page_chunks` doesn't give even sized chunks, the first chunk will be of size remainder. If someone wants even sized chunks of a page they could use `rechunk_event_pages`. Are there some uses for a chunking function that has a remainder argument?